docs/root/configuration/secret.rst

```diff
@@ -0,0 +1,146 @@
+
+Examples two: SDS server
+------------------------
+
+This example shows how to configurate secrets fetched from remote SDS server:
+
+.. code-block:: yaml
+
+    clusters:
+      - name: sds_server_mtls
+        http2_protocol_options: {}
+        hosts:
+          socket_address:
+            address: 127.0.0.1
+            port_value: 8234
+        tls_context:
+          common_tls_context:
+          - tls_certificate:
+            certificate_chain:
+              filename: certs/sds_cert.pem
+            private_key:
+              filename: certs/sds_key.pem
+      - name: sds_server_uds
+        http2_protocol_options: {}
+        hosts:
+          - pipe:
+              path: /tmp/uds_path
+      - name: example_cluster
+        connect_timeout: 0.25s
+        hosts:
+        - name: local_service_tls
+          ...
+          tls_context:
+            common_tls_context:
+              tls_certificate_sds_secret_configs:
+              - name: "client_cert"
+                sds_config:
+                  api_config_source:
+                    api_type: GRPC
+                    grpc_services:
+                      google_grpc:
+                        target_uri: unix:/tmp/uds_path
+    listeners:
+      ....
+      filter_chains:
+        tls_context:
+          common_tls_context:
+            tls_certificate_sds_secret_configs:
+            - name: "server_cert"
+              sds_config:
+                api_config_source:
+                  api_type: GRPC
+                  grpc_services:
+                    envoy_grpc:
+                      cluster_name: sds_server_mtls
+            validation_context_sds_secret_config:
+              name: "validation_context"
+              sds_config:
+                api_config_source:
+                  api_type: GRPC
+                  grpc_services:
+                    envoy_grpc:
+                      cluster_name: sds_server_uds
+
+
+For illustration, above example uses 3 ways to access the SDS server. A gRPC SDS server can be reached by Unix Domain Socket path "/tmp/uds_path" and "127.0.0.1:8234" by mTLS. It can provide three secrets, "client_cert", "server_cert" and "validation_context". In the config, cluster "example_cluster" certificates "client_cert" is configured to use google_grpc by using UDS to talk to SDS server. The Listener needs to fetch "server_cert" and "validation_context" from the SDS server. The "server_cert" is using envoy_grpc with cluster "sds_server_mtls" configured with client certificates to use mTLS to talk to SDS server. The "validate_context" is using envoy_grpc with cluster "sds_server_uds" configured with UDS path to talk to the SDS server.
```

Comment:
Nit: can you cleanup the formatting of fields, values and gRPC types here to make it look pretty and consistent with the rest of Envoy docs? Thanks.

Reply:
Done